### PR TITLE
Runnable recipe: introduce validation and resolver

### PR DESCRIPTION
**Pull request by clebergnu**
The concepts of recipes were pretty much internal to Avocado.  Now, there are use cases for relying on them as a data file format.  For this reason, let's introduce a validation of the current JSON file format.  In the future, we may evaluate whether it needs to be further specialized, like adding signatures/magic numbers, specific file extensions, etc.

This also introduces a resolver implementation.  With the validation introduced here, runnable recipes (JSON files) that are well formed and contain the right data, will be turned into internal runnables that can be listed with `avocado list` and run with `avocado run`.

### CODE REVIEW

**Review by richtja — CHANGES_REQUESTED** [review]
Hi @clebergnu, I tested it and it LGTM. But since we are opening runnable recipes to users, I think we need to describe it in avocado documentation otherwise it will stay hidden, and it will be only used by pasta.

File: docs/source/guides/writer/chapters/recipes.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
+Defining what to run using recipe files
+---------------------------------------
+
+If you've followed the previous documentation sections, you should now
+be able to write ``exec-test`` tests and also ``avocado-instrumented``
+tests.  These tests you should be found when you run
```
Comment (richtja) [inline docs/source/guides/writer/chapters/recipes.rst:6]: typo These tests you should be found/These tests should be found

File: docs/source/guides/writer/chapters/recipes.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+.. literalinclude:: ../../../../../avocado/schemas/runnable-recipe.schema.json
+
+Avocado will attempt to enforce the JSON schema any time a
+``Runnable`` is loaded from such recipe files.
+
+Using runnable recipes as references
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Avocado ships with a ``runnable-recipe`` resolver plugin, which means
+that you can use runnable recipe fileas as a reference, and get
```
Comment (richtja) [inline docs/source/guides/writer/chapters/recipes.rst:40]: typo: fileas/files

File: docs/source/guides/writer/chapters/recipes.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
+Defining what to run using recipe files
+---------------------------------------
+
+If you've followed the previous documentation sections, you should now
+be able to write ``exec-test`` tests and also ``avocado-instrumented``
+tests.  These tests you should be found when you run
+``avocado run /reference/to/a/test``.  Internally, though, these will
+be describe as a :class:`avocado.core.nrunner.runnable.Runnable`.
```
Comment (richtja) [inline docs/source/guides/writer/chapters/recipes.rst:8]: typo: describe/described

**Review by richtja — CHANGES_REQUESTED** [review]
Hi @clebergnu, the additional documentation LGTM, I just found few typos.

File: docs/source/guides/writer/chapters/recipes.rst (hunk 3 above)
Comment (clebergnu) [reply docs/source/guides/writer/chapters/recipes.rst:8]: I actually meant "defined".  Changing it here, thanks!

**Review by richtja — APPROVED** [review]

**Pull request merged**